script.js

```diff
@@ -1,72 +1,85 @@
-let booksList = [];
+class Book {
+  constructor(title, author) {
+    this.id = Math.random() * 1000;
+    this.title = title;
+    this.author = author;
+  }
 
-const form = document.querySelector('.add-book');
-const books = document.querySelector('.books');
-const title = document.getElementById('title');
-const author = document.getElementById('author');
+  storeBook() {
+    let book = {};
+    if (this.title && this.author !== '') {
+      book = {
+        id: this.id,
+        title: this.title,
+        author: this.author,
+      };
+    }
+
+    return book;
+  }
 
-// add the book to the view page
-const addBook = (book) => {
-  const bookItem = document.createElement('p');
-  bookItem.setAttribute('id', book.id);
-  const titlePara = document.createElement('p');
-  titlePara.textContent = book.title;
-  const authorPara = document.createElement('p');
-  authorPara.textContent = book.author;
+  addBook() {
+    const bookItem = document.createElement('tr');
+    bookItem.setAttribute('id', this.id);
+    const titlePara = document.createElement('td');
+    titlePara.textContent = `${this.title} by ${this.author}`;
 
-  const lineBreak = document.createElement('hr');
+    const books = document.querySelector('.books');
 
-  function removeBook(bookId) {
-    const newArray = booksList.filter((book) => book.id !== bookId);
-    localStorage.setItem('booksList', JSON.stringify(newArray));
+    const deleteButton = document.createElement('button');
+    deleteButton.setAttribute('type', 'button');
+    deleteButton.setAttribute('class', 'delete');
+    deleteButton.textContent = 'Remove';
 
-    const booksId = document.getElementById(bookId);
-    booksId.remove();
+    titlePara.appendChild(deleteButton);
+    bookItem.appendChild(titlePara);
 
-    booksList = newArray;
+    deleteButton.addEventListener('click', () => this.removeBook(this.id));
+
+    return books.appendChild(bookItem);
   }
 
-  const deleteButton = document.createElement('button');
-  deleteButton.setAttribute('type', 'button');
-  deleteButton.setAttribute('class', 'delete');
-  deleteButton.textContent = 'Remove';
-  deleteButton.addEventListener('click', () => removeBook(book.id));
-
-  bookItem.appendChild(titlePara);
-  bookItem.appendChild(authorPara);
-  bookItem.appendChild(deleteButton);
-  bookItem.appendChild(lineBreak);
-
-  books.appendChild(bookItem);
-};
-
-// load books from local storage
-const loadBooks = () => {
-  const storedBooks = localStorage.getItem('booksList');
-  if (storedBooks !== null) {
-    booksList = JSON.parse(storedBooks);
-    booksList.forEach((book) => {
-      addBook(book);
-    });
+  removeBook(id) {
+    const newArray = JSON.parse(localStorage.getItem('booksList'));
+    newArray.filter(() => this.id !== id);
+
+    const booksId = document.getElementById(id);
+    return booksId.remove();
```

Comment:
- [ ] The remove book function only removes the book from the DOM, it doesn't remove the book from the `booklist` or the localStorage. please make sure to update your book array when removing a book.

Reply:
well resolved! ✔️